provider/pkg/provider/http.go

```diff
@@ -207,3 +236,39 @@ func addQueryParams(rawurl string, params map[string]string) (string, error) {
 	u.RawQuery = q.Encode()
 	return u.String(), nil
 }
+
+// staticTokenSource is used to be able to identify static token sources without reflection.
+type staticTokenSource struct {
+	oauth2.TokenSource
+}
+
+// If the argument is a path, pathOrContents loads it and returns the contents,
+// otherwise the argument is assumed to be the desired contents and is simply
+// returned.
+//
+// The boolean second return value can be called `wasPath` - it indicates if a
+// path was detected and a file loaded.
+func pathOrContents(poc string) (string, bool, error) {
+	if len(poc) == 0 {
+		return poc, false, nil
+	}
+
+	path := poc
+	if path[0] == '~' {
+		var err error
+		path, err = homedir.Expand(path)
+		if err != nil {
+			return path, true, err
```

Comment:
Maybe should return an empty string for path to make it clearer that there was an error?